sunpy/map/maputils.py

```diff
@@ -1,7 +1,268 @@
+    rhs = list(product(np.arange(ny), [nx - 1])) * u.pix
+    return {"top": top, "bottom": bottom, "lhs": lhs, "rhs": rhs}
+
+
+def contains_full_disk(smap):
+    """
+    Checks if a map contains the full disk of the Sun.
+
+    A map contains the full disk of the Sun if the following two
+    conditions are met: (1) all the pixels at the edge of the map are
+    more than 1 solar radius from the center of the Sun and, (2) the
+    map is not all off disk. If both these conditions are met, the
+    function returns `True`. Otherwise, the function returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `False` if any of the edge pixels are less than one solar
+        radius away from the center of the Sun.
+
+    Notes
+    -----
+    * The disk itself need not be imaged.  For example, in coronagraph images such as those
+    * from LASCO C2 and C3 the full disk is within the field of view of the instrument,
+    * but the solar disk itself is not imaged.
+
+    """
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges["lhs"], edges["rhs"], edges["top"], edges["bottom"]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    edge_of_world = smap.pixel_to_world(x, y)
+
+    # Calculate the distance of the edge of the world in solar radii
+    distance = u.R_sun * np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2) / smap.rsun_obs
+
+    # Test if all the edge pixels are more than one solar radius distant
+    # and that the whole map is not all off disk.
+    return np.all(distance > 1*u.R_sun) and ~is_all_off_disk(smap)
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's angular distance
+    to the angular size of the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if the coordinate is on disk, `False` otherwise.
+    """
+    # Calculate the radius of every pixel from the center of the Sun
+    # in helioprojective Cartesian coordinates.
+    coordinate_angles = np.sqrt(coordinate.Tx ** 2 + coordinate.Ty ** 2)
+    # The tangent of the angular size of the solar radius is equal to the radius
+    # of the Sun divided by the distance between the observer and the
+    # center of the Sun.
+    solar_radius_angular_size = np.arctan(coordinate.rsun / coordinate.observer.radius.to(u.km)).to(u.arcsec)
```

Comment:
Do you need the `.to(u.km)` here?

Reply:
You're right.  The Units are handled automagically.  Great!